src/main/java/nl/tudelft/jpacman/npc/ghost/Ghost.java

```diff
@@ -16,27 +16,106 @@
  * @author Jeroen Roosen 
  */
 public abstract class Ghost extends NPC {
-	
+
+	/**
+	 * Strategy used by ghosts to choose which way to go.
+	 */
+	protected Strategy strategyMode;
 	/**
 	 * The sprite map, one sprite for each direction.
 	 */
 	private Map<Direction, Sprite> sprites;
 
+	private Square home;//Sa case maison
+	public Direction[] chemin;
+	public Direction[] cheminEnCours;//Ses prochaines directions à prendre lors du tour dans la maison
+	public boolean atteintHome;//S'il a atteint sa case maison
+	
 	/**
 	 * Creates a new ghost.
 	 * 
 	 * @param spriteMap
 	 *            The sprites for every direction.
 	 */
+	protected Ghost(Map<Direction, Sprite> spriteMap,String strategy) {
+		super(strategy);
+		this.sprites = spriteMap;
+	}
+
 	protected Ghost(Map<Direction, Sprite> spriteMap) {
 		this.sprites = spriteMap;
 	}
 
+	protected Ghost(Map<Direction, Sprite> spriteMap, boolean aH, String strategy) {
+		super(strategy);
+		this.sprites = spriteMap;
+		this.atteintHome = aH;
+	}
+
+	public void setStrategyMode(Strategy strategyMode) {
+		this.strategyMode = strategyMode;
+	}
+	public abstract void setPoursuiteMode();
+	public abstract void setDispersionMode();
+
+
 	@Override
 	public Sprite getSprite() {
 		return sprites.get(getDirection());
 	}
 
+	public Square getHome(){
+		return this.home;
+	}
+	
+	public void setHome(Square home){
+		this.home=home;
+	}
+	
+	public String getStrategy(){
+		return this.strategy;
+	}
+
+	public void setStrategy(String strategy){
+		if (this.getStrategy() != strategy){
+			if (strategy == "modePoursuite" && this.atteintHome){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare strings instead of '==' or '!='](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2533370832/issues/source?bid=3267846&fileBranchId=3284210#l81)